### ISSUE TITLE
computing _u in planar flow

### ISSUE THREAD

**Issue by BigTuna08** [LukasRinder/normalizing-flows#2]
I think in the self._u() method of your PlanarFlow class, you have

tf.math.sqrt(tf.reduce_sum(self.w ** 2.0)))

but it should be just 
tf.reduce_sum(self.w ** 2.0))

as the paper uses the squared norm.

**Comment by FabricioArendTorres** [comment]
Aside from that issue, this code does not restrict the u, so that the layer is invertible (w^t@u >-1).
Instead, the planar flow implementation basically reprojects the u, if w^t@u>-1 once it is not fulfilled anymore.
